pallets/claim/src/lib.rs

```diff
@@ -0,0 +1,467 @@
+                Fortitude::Polite,
+            )
+        }
+
+        fn do_claim(origin: T::AccountId, beneficiary: Option<T::AccountId>) -> DispatchResult {
+            // See if account is eligible to get an airdrop
+            Beneficiaries::<T>::try_mutate_exists(origin.clone(), |amount| {
+                *amount = match amount {
+                    // Account is eligible to get an airdrop
+                    Some(amount) => {
+                        // Determine who is the beneficiary
+                        let beneficiary = beneficiary.unwrap_or(origin);
+                        // Execute payment
+                        let available = Self::pot();
+                        if *amount > available {
+                            log::warn!("Claimable amount {amount:?} bigger than total available {available:?}");
+                            Err(Error::<T>::PayoutError)?; // Prevent going under the existential deposit of the account
+                        }
+                        let payment_id = T::Paymaster::pay(&beneficiary, (), *amount)
+                            .map_err(|_| Error::<T>::PayoutError)?;
+                        // Subtract the claimed token from the TotalClaimable
+                        TotalClaimable::<T>::mutate(|required_amount| {
+                            *required_amount = required_amount.saturating_sub(*amount)
+                        });
+                        log::trace!("Claimed {amount:?} for {beneficiary:?}");
+                        Self::deposit_event(Event::<T>::Claimed {
+                            beneficiary,
+                            amount: *amount,
+                            payment_id,
+                        });
+                        None
+                    }
+                    // Account is not eligible to receive funds
+                    _ => Err(Error::<T>::NotEligible)?,
+                };
+                Ok::<_, DispatchError>(())
+            })?;
+            Ok(())
+        }
+
+        fn do_add_beneficiaries(
+            beneficiaries: BTreeMap<T::AccountId, BalanceOf<T>>,
+        ) -> DispatchResult {
+            // Check that the pot has enough funds to cover for all the beneficiaries
+            let available_amount = Self::pot();
+            let mut required_amount = TotalClaimable::<T>::get();
+
+            beneficiaries
+                .iter()
+                .try_for_each::<_, DispatchResult>(|(account, amount)| {
+                    if let Some(_) = Beneficiaries::<T>::get(account.clone()) {
+                        // Account already exists
+                        log::warn!("Beneficiary {account:?} already added.");
+                        Err(Error::<T>::AlreadyPresent)?;
+                    } else if amount.is_zero() {
+                        // Attempting to add a beneficiary with nothing to claim
+                        log::warn!("Beneficiary {account:?} with nothing to claim.");
```

Comment:
Use guard approach instead:

```rust
if Beneficiaries::<T>::contains_key(account) {
    // Account already exists
    log::warn!("Beneficiary {account:?} already added.");
    Err(Error::<T>::AlreadyPresent)?;
} 
if amount.is_zero() {
    // Attempting to add a beneficiary with nothing to claim
    log::warn!("Beneficiary {account:?} with nothing to claim.");
    Err(Error::<T>::NothingToClaim)?;
}
// Account doesn't exist. Add its token amount to the required amount this pallet's account should have
required_amount = required_amount.saturating_add(*amount);
log::trace!("Added beneficiary {account:?}. Can claim: {amount:?}");
```

It's not _structured_ programming but is nicer and more clear to understand: the business logic is in the main branch code